dspace-spring-rest/src/main/java/org/dspace/app/rest/MappingItemRestController.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.log4j.Logger;
+import org.dspace.app.rest.converter.ItemConverter;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.model.ItemRest;
+import org.dspace.app.rest.model.MappingItemRestWrapper;
+import org.dspace.app.rest.model.hateoas.MappingItemResourceWrapper;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.content.Collection;
+import org.dspace.content.Item;
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/core/collections/" +
+    "{uuid:[0-9a-fxA-FX]{8}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{12}}/mappingItems")
```

Comment:
It seems odd to me that this endpoint is `/collections/[uuid]/mappingItems`, while the Item endpoint is `/items/[uuid]/mappedCollections`.  Shouldn't these be named similarly?

I'd recommend renaming this to be `/collections/[uuid]/mappedItems`.  I realize that may also require minor changes to the REST Contract, but it seems more consistent

Reply:
Hi @tdonohue thank you for the feedback. That would indeed make a lot more sense and I'll change this in the following commit. However I had an additional question as it may make sense to rename everything from 'mappingItems' to 'mappedItems' then if we're changing the endpoint syntax. For reference: https://github.com/DSpace/DSpace/pull/2282/files#diff-c56be5b7c98dee81bed835af5bf7cb6dR45
Currently they get embedded under the 'mappingItems' title, should we change this to 'mappedItems' as well along with refactoring class names etc? Just to make it more consistent with the proposed change as that would make everything even more consistent

Reply:
@Raf-atmire : Yes, I agree and I'd be in favor of renaming all "mappingItems" to "mappedItems" for consistency.